### PR TITLE
Delete partitions which are older than threshold days

### PR DESCRIPTION
**Pull request by msvinaykumar**
Remove partitions that exceed the specified threshold days.

### CODE REVIEW

File: src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java [hunk 1]
```diff
@@ -75,6 +78,47 @@ public ValidationOutputData addExperimentToDB(KruizeExperimentEntry kruizeExperi
...
+                if (tableName.startsWith("kruize_results_")) {
+                    datePart = tableName.substring("kruize_results_".length());
+                } else if (tableName.startsWith("kruize_recommendations_")) {
+                    datePart = tableName.substring("kruize_recommendations_".length());
+                }
+                if (null != datePart) {
+                    Date tableDate = new SimpleDateFormat("yyyyMMdd").parse(datePart);
+                    // Compare the date part with the cutoff date
+                    if (tableDate.after(cutoffDate)) {
+                        LOGGER.info("Table not eligible for deletion: " + tableName);
```
Comment (khansaad) [inline src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java:106]: Should we set it as `debug` ?

File: src/main/java/com/autotune/utils/KruizeConstants.java [hunk 2]
```diff
@@ -417,6 +417,7 @@ public static final class KRUIZE_CONFIG_ENV_NAME {
         public static final String AUTOTUNE_MODE = "autotunemode";
         public static final String EM_ONLY_MODE = "emonly";
         public static final String BULK_UPDATE_RESULTS_LIMIT = "bulkresultslimit";
+        public static final String DELETE_PARTITION_THRESHOLD_IN_DAYS = "deletepartitionsthreshold";
```
Comment (khansaad) [inline src/main/java/com/autotune/utils/KruizeConstants.java:420]: Are we using this anywhere ?

File: src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java (hunk 1 above)
Comment (msvinaykumar) [reply src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java:106]: It's a less log, and I want to confirm whether the correct table is being deleted.

File: src/main/java/com/autotune/operator/KruizeDeploymentInfo.java [hunk 3]
```diff
@@ -66,14 +66,12 @@ public class KruizeDeploymentInfo {
     public static Boolean settings_save_to_db;
     public static String em_only_mode;
     public static Integer bulk_update_results_limit = 100;
-
     public static int generate_recommendations_date_range_limit_in_days = 15;
-
+    public static String delete_partition_threshold_in_days;
```
Comment (chandrams) [inline src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: Can you please update the docs on how to use this?

Comment (chandrams) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: Has this been tested?

Comment (msvinaykumar) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: yes on [EE](https://console-openshift-console.apps.c-rh-c-eph.8p0c.p1.openshiftapps.com/k8s/cluster/projects/ephemeral-ujnc03) and created [MR](https://github.com/RedHatInsights/ros-ocp-backend/pull/163) too

Comment (msvinaykumar) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: > Can you please update the docs on how to use this?

done

Comment (chandrams) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: Please include the test into our kruize tests to validate both default and custom threshold value

Comment (msvinaykumar) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: @chandrams Could you kindly provide a reference test suite for verifying default and custom values set via environment variables? Or, is a new test suite needed for this purpose? Also, how are other params unrelated to this PR being tested?

Comment (chandrams) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: We do not have any tests related to environment values.

Comment (msvinaykumar) [reply src/main/java/com/autotune/operator/KruizeDeploymentInfo.java:70]: Chandrakal, appreciate it. This appears to be a significant change, requiring a framework for assessment. Should we contemplate creating a distinct PR to address the broader scope?

**Comment by msvinaykumar** [comment]
THis PR merged to mvp_demo by creating new PR
https://github.com/kruize/autotune/pull/1084

**Pull request closed without merging**